accesspoc/dips/forms.py

```diff
@@ -98,3 +101,38 @@ def save(self, commit=True):
         if commit:
             user.save()
         return user
+
+
+class DublinCoreSettingsForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        super(forms.Form, self).__init__(*args, **kwargs)
+        self.optional_fields_setting = Setting.objects.get(name='enabled_optional_dc_fields')
+        self.fields['optional_fields'] = forms.MultipleChoiceField(
+            required=False,
+            label=_('Optional fields'),
+            help_text=_('Ctrl + Click to select multiple or unselect.'),
+            choices=DublinCore.get_optional_fields().items(),
+            # Convert JSON string to list to set initial data
+            initial=json.loads(self.optional_fields_setting.value),
+        )
+        self.hide_empty_fields_setting = Setting.objects.get(name='hide_empty_dc_fields')
+        self.fields['hide_empty_fields'] = forms.BooleanField(
+            required=False,
+            label=_('Hide empty fields'),
+            help_text=_(
+                'Check to not display Dublin Core fields without data '
+                'in the Collection and Folder view pages.'
+            ),
+            # Convert string to boolean to set initial data
+            initial=str_to_bool(self.hide_empty_fields_setting.value),
+        )
+
+    def save(self):
+        # Convert list to JSON string to save in database
+        value = json.dumps(self.cleaned_data['optional_fields'])
+        self.optional_fields_setting.value = value
+        self.optional_fields_setting.save()
+        # Convert boolean to string to save in database
+        value = str(self.cleaned_data['hide_empty_fields'])
+        self.hide_empty_fields_setting.value = value
+        self.hide_empty_fields_setting.save()
```

Comment:
It'd be nicer if you could just do something like the following:

    for name, field in self.fields.items():
        self.instances[name].value = self.cleaned_data[name]
        self.instances[name].save()

This would be possible if you stopped treating booleans differently, i.e. you can also encode a boolean as a JSON literal:

```python
>>> import json
>>> json.dumps(True)
'true'
>>> json.loads('true')
True
>>> json.dumps({1: 2})
'{"1": 2}'
```